activestorage/app/controllers/active_storage/disk_controller.rb

```diff
@@ -34,8 +34,8 @@ def update
   end
 
   private
-    def disk_service
-      ActiveStorage::Blob.service
+    def named_disk_service(name)
+      ActiveStorage::Blob.services.fetch(name)
```

Comment:
@georgeclaghorn How should we handle adding new attributes on encoded keys? This will break any signed keys that were generated _before_ this change is deployed to a rails app. I think we can probably just revert to the old behaviour in the event that we don't have a service name.
```suggestion
      ActiveStorage::Blob.services.fetch(name) do
        ActiveStorage::Blob.service
      end
```

Reply:
That seems fine. :+1: